From 8bf82ba6095219f8502afb022af6484b2ee18df6 Mon Sep 17 00:00:00 2001
From: Han Xu <han.xu@nxp.com>
Date: Thu, 7 Dec 2017 14:28:34 -0600
Subject: [PATCH 3001/5242] MLK-17120: arm64: dts: assign the clock rate for
 GPMI NAND in DT

commit  b5e91c15d7eabaac50ddb7d0c0fe7ce45504ee01 from
https://source.codeaurora.org/external/imx/linux-imx.git

Assign the clock rate for GPMI NAND on i.MX8QXP ARM2 device tree.

To keep the same clock rate after system suspend/resume, we need to set
assign a clock rate for GPMI NAND, otherwise the timing register won't
match with the clock setting.

The code change also a workaround for SCU clock rate setting. NAND use a
very low clock freq (22Mhz) and safe timing to identify which chips were
connected. This low freq divide from high freq parent clock(1Ghz) caused
the SCU clock divider go beyond the limit (31)

SCU need to implement the clk_round_rate to found this issue and return
error value to upper layer. Right now assign 50Mhz for GPMI initial
clock as a workaround.

Signed-off-by: Han Xu <han.xu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi
index 9880c11..fa0bbdf 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp.dtsi
@@ -1603,6 +1603,8 @@
 		dmas = <&dma_apbh 0>;
 		dma-names = "rx-tx";
 		power-domains = <&pd_conn_nand>;
+		assigned-clocks = <&clk IMX8QXP_GPMI_BCH_IO_CLK>;
+		assigned-clock-rates = <50000000>;
 		status = "disabled";
 	};
 
-- 
1.7.9.5

